src/librustdoc/clean/mod.rs

```diff
@@ -599,14 +596,14 @@ impl Clean<GenericParamDef> for hir::GenericParam<'_> {
                     for bound in bounds {
                         s.push_str(&format!(" + {}", bound.name.ident()));
                     }
-                    s
+                    Symbol::intern(&s)
                 } else {
-                    self.name.ident().to_string()
+                    self.name.ident().name
```

Comment:
Ah ok, this is where you needed 'either a symbol or a string', because it can be calculated on demand. Yeah, I think interning is the right approach here - then if you generate the same string twice it doesn't use more memory.